Initial commit of localectl role

tehsmyers opened the pull request:
Two scenarios are included:
- `default` represents a "normal" run with idempotence
- `conditional_tests`, as the name implies, ensures that all the
  conditionals in `tasks/main.yml` end up executing tasks as intended.

Closes #3

tehsmyers commented:
Various changes to the meta_skeleton happened here:
- oasis-roles/oasis_roles in the README example playbook (I already [ninja-patched](https://github.com/oasis-roles/meta_skeleton/commit/6d382b970be0247b30f65e520f4bb55345dc46b6) meta_skeleton for this one)
- python 3 version updated to 3.7 in travis; molecule will [fail loudly](https://github.com/ansible/molecule/blob/master/molecule/shell.py#L75-L78) if run on python 3.5, and I see no reason to use 3.6 if 3.7 is available for testing.
- `flake8` call removed from travis scripts, as molecule does this for us (or am I missing something?)

I have opened up https://github.com/oasis-roles/meta_skeleton/pull/14 with these changes, and for additional discussion.

tehsmyers commented:
~~I kinda sorta completely forgot to add the `become` vars and handling, so just pretend that's there for now. :)~~

The need to pretend has been removed.

greg-hellings commented on .travis.yml:
```diff
@@ -18,9 +21,7 @@ install:
-  - molecule lint -s openstack
-  - molecule test -s docker
-  - flake8 $(find . -name '*.py')
+  - molecule test -s $MOLECULE_SCENARIO
```
Shell variables should be quoted!!! `"${MOLECULE_SCENARIO}"`

Like, ShellCheck would say that they should but it's probably just fine here.

tehsmyers commented:
We've also updated meta_skeleton to decrease the likelihood of forgetting the `become` vars in the future.

greg-hellings commented on README.md:
```diff
@@ -33,16 +52,160 @@ Example Playbook
+Virtual Console Locale Setting Notes
+------------------------------------
+
+Valid virtual console keymap setting values can be determind by running
```
"determind"

robled commented on defaults/main.yml:
```diff
@@ -1 +1,18 @@
-# defaults file for localectl
+# "become" handling
+localectl_become: true
```
Is `localectl_become_user` not needed for this role?

tehsmyers replied:
You can become any user you want, as long as that user is `root`.

robled commented on filter_plugins/localectl_filters.py:
```diff
@@ -0,0 +1,103 @@
+The filters in this module serve to simplify this role's tasks by formatting
+busctl's output into objects usable in ansible tasks, and to make reasonable
+decisions for the purposes of idempotence. This could all be done as "pure"
+ansible tasks, but those tasks would be nigh unredable.
```
*unreadable (meta++)

robled commented on tasks/main.yml:
```diff
@@ -1 +1,71 @@
+    {{ localectl_properties | join(' ') }}
+  register: localectl_busctl_gather
+  changed_when: false
+  become: "{{ localectl_become }}"
```
Does this require elevated privileges to run?  On my system I can grab info with `busctl` using a non-privileged account.

tehsmyers replied:
I had mixed experiences with getting information from the system dbus stuff. The short answer is "I'm not sure, so I'm playing it safe."

tehsmyers replied:
That said, I'll remove it, since this is why we have CI, and my personal experience is effectively worthless as long as it works in CI.

robled commented on tasks/main.yml:
```diff
@@ -1 +1,71 @@
+    localectl_facts: |
+      {{ localectl_busctl_gather.stdout_lines |
+         localectl_busctl_parse(localectl_properties) }}
+  become: "{{ localectl_become }}"
```
Does setting facts require `become`?

tehsmyers replied:
No, but it doesn't require an ssh connection either, so this is pretty funny.

greg-hellings commented on README.md:
```diff
@@ -33,16 +52,160 @@ Example Playbook
+
+Contrary to the `localectl` default behavior, a value given for
+`localectl_vc_keymap` will not automatically be converted to
+`localectl_x11_layout`. See the "`--no-convert` behavior`" section below
```
There's an extra "`" character after the word "behavior" on this line.

greg-hellings commented on README.md:
```diff
@@ -33,16 +52,160 @@ Example Playbook
+
+Contrary to the `localectl` default behavior, a value given for
+`localectl_x11_layout` will not automatically be converted to
+`localectl_vc_keymap`. See the "`--no-convert` behavior`" section below
```
Repeat of the extra back tick here.

tehsmyers replied:
Delicious copypasta

greg-hellings commented on README.md:
```diff
@@ -33,16 +52,160 @@ Example Playbook
+`options` being unset.
+
+To prevent unexpected results, explicitly defining all `localectl_vc_*` or
+`localectl_x11_* vars is recommended to ensure the intended changes are made
```
Missing the closing backtick on this variable name.

greg-hellings commented on filter_plugins/localectl_filters.py:
```diff
@@ -0,0 +1,103 @@
+    # could also 'for value in propvalues' here, but this makes use of the
+    # length in the return string from busctl
+    for i in range(proplen):
+        # strip off the quotes, as in localectl_busctl_s
```
Why are you not using `localectl_busctl_s` here?

tehsmyers replied:
Addressed by expanding the comment to answer that question.

greg-hellings commented on molecule/default/INSTALL.rst:
```diff
@@ -0,0 +1,22 @@
+*******
```
Is this a standard part of the Molecule template? I don't recall having `.rst` files.

tehsmyers replied:
It is, I just missed this one when adding the new scenario. I think it might be trivial to make molecule use a different template dir than the one baked into molecule itself, so using `molecule init scenario` like I did here wouldn't result in too many surprises like this.

tehsmyers replied:
File removed.

greg-hellings commented on tasks/main.yml:
```diff
@@ -1 +1,69 @@
+  become: "{{ localectl_become }}"
+  when: |
+    localectl_locales is not none and
+    localectl_facts['Locale'] != localectl_locales
```
Does Jinja properly handle equality of dicts, item by item?

tehsmyers replied:
I don't think it does, in that it reverts to python dict equality. Python dict equality does handle this correctly.

I think I'll go ahead and throw some `| dictsort`s on there anyway, and get it covered with a test, so we're guarded against anyone, anywhere, deciding to do something that results in us seeing anything like an `OrderedDict` here.

tehsmyers replied:
I put this up in its own commit, so it's easy to rip out if we don't like it:
https://github.com/oasis-roles/localectl/pull/5/commits/1b65bbe9c8623e1282f075003bf19727f7986c7d

Commit:
Initial commit of localectl role

Two scenarios are included:
- `default` represents a "normal" run with idempotence
- `conditional_tests`, as the name implies, ensures that all the
  conditionals in `tasks/main.yml` end up executing tasks as intended.

Signed-off-by: Sean Myers <sean.myers@redhat.com>

Commit:
Sort dicts before compare, and test dict compare

This might be overkill, so I'm putting it in its own separate commit in
case it's decided in review that we don't want it.

Signed-off-by: Sean Myers <sean.myers@redhat.com>
